### ISSUE TITLE
Different activation for each hidden layer in Functional

### ISSUE THREAD

**Issue by pradhyumna85** [sciann/sciann#43]
How to provide different activation for each hidden layer in the Functional instead of the same activation for all hidden layers?

**Comment by ehsanhaghighat** [comment]
Used to be possible by passing a list of activation funcs of the same size as the len of hidden layers. 
But now if you pass a list, it gets treated as Rowdy net. 
You can still build such a network but you need to do it layer by layer. 
Why do you need this? It is generally not recommended.

**Comment by pradhyumna85** [comment]
Hi Ehsan,
Thank you for your quick response and sorry for my late reply.
Actually, the reason I want to try this is that I am trying to replicate the results of this paper : [Here](https://www.researchgate.net/publication/355431293_Physics_Inspired_Machine_Learning_for_Solving_Fluid_Flow_in_Porous_Media_A_Novel_Computational_Algorithm_for_Reservoir_Simulation). According to this paper having mixed activations help in convergence in case of a 2D laplacian PDE with source and sinks (basically discontinuities/shocks in the solution).

Anyway, I modified the code of sciann.Functional to accept a list of activations and after trying this it hugely improved the results.

Btw, you mentioned that such network can be built layer by layer, I'll be glad if you could share a small example please.🙂

**Comment by pradhyumna85** [comment]
Hi @ehsanhaghighat,
I think it will be good to have a option to pass list of activations in the sciann.functional, just for flexibility (and I have tested that mixed activation does help in convergence) and a parameter, something like "rowdy_net" (True/False) can also be added which would be applicable only if list of activations are passed.
Let me know your thoughts on this, if its a green light from you then I'll create a pull request.

Thank you.

**Comment by Mengxiangqi0** [comment]
> Hi @ehsanhaghighat, Thank you for your quick response and sorry for my late reply. Actually, the reason I want to try this is that I am trying to replicate the results of this paper : [Here](https://www.researchgate.net/publication/355431293_Physics_Inspired_Machine_Learning_for_Solving_Fluid_Flow_in_Porous_Media_A_Novel_Computational_Algorithm_for_Reservoir_Simulation). According to this paper having mixed activations help in convergence in case of a 2D laplacian PDE with source and sinks (basically discontinuities/shocks in the solution).
> 
> Anyway, I modified the code of sciann.Functional to accept a list of activations and after trying this I found out that it does seem to hugely improve the results.
> 
> Btw, you mentioned that such a network can be built layer by layer, I'll be glad if you could share a small example, please.🙂

Hello, sorry to bother you. I also have this issue now. I would like to add different activation functions to my hidden layer, as I have seen in the literature that this strategy seems to lower the loss and weaken the spectral deviation problem of PINN. I would like to ask how you implemented this in SCIANN?
Good luck!
